wgsl/index.bs

```diff
@@ -1701,57 +1712,74 @@ required alignment of host-shareable type |S| when used by storage class |C|.
+  <tr><td>atomic&lt;T&gt;
+      <td>[=AlignOf=](|S|)
+      <td>[=AlignOf=](|S|)
+  <tr><td>vecN&lt;T&gt;
       <td>[=AlignOf=](|S|)
       <td>[=AlignOf=](|S|)
   <tr algorithm="alignment of a matrix with N columns and M rows">
-      <td>mat|N|x|M|&lt;f32&gt;
+      <td>matNxM&lt;f32&gt;
       <td>[=AlignOf=](|S|)
       <td>[=AlignOf=](|S|)
   <tr algorithm="alignment of an array">
-      <td>array<|T|,|N|>
-      <td>[=AlignOf=](|T|)
-      <td>[=roundUp=](16, [=AlignOf=](|T|))
+      <td>array&lt;T, N&gt;
+      <td>[=AlignOf=](|S|)
+      <td>[=roundUp=](16, [=AlignOf=](|S|))
   <tr algorithm="alignment of an runtime-sized array">
-      <td>array<|T|>
-      <td>[=AlignOf=](|T|)
-      <td>[=roundUp=](16, [=AlignOf=](|T|))
+      <td>array&lt;T&gt;
+      <td>[=AlignOf=](|S|)
+      <td>[=roundUp=](16, [=AlignOf=](|S|))
   <tr algorithm="alignment of a structure">
-      <td>struct&lt;T<sub>0</sub>, ..., T<sub>N</sub>&gt;
-      <td>max([=AlignOf=](T<sub>0</sub>), ..., [=AlignOf=](T<sub>N</sub>))
-      <td>[=roundUp=](16, max([=AlignOf=](T<sub>0</sub>), ..., [=AlignOf=](T<sub>N</sub>)))<br>
-  <tr><td>atomic<|T|>
-      <td>[=AlignOf=](|T|)
-      <td>[=AlignOf=](|T|)
+      <td>struct |S|
+      <td>[=AlignOf=](|S|)
+      <td>[=roundUp=](16, [=AlignOf=](|S|))<br>
 </table>
 
-All structure members of type |T| must have a byte offset from the start of the
-structure that is a multiple of the [=RequiredAlignOf=](|T|, |C|) for the storage
-class |C|:
+Structure members of type |T| must have a byte offset
+from the start of the structure that is a multiple of the [=RequiredAlignOf=](|T|, |C|)
+for the storage class |C|:
 
 <p algorithm="structure member minimum alignment">
-    [=OffsetOf=](|S|, |M|) = |k| &times; [=RequiredAlignOf=](|T|, C)<br>
+    [=OffsetOfMember=](|S|, |M|) = |k| &times; [=RequiredAlignOf=](|T|, C)<br>
     Where |k| is a non-negative integer and |M| is a member of structure |S| with type |T|
 </p>
 
-All arrays of element type |T| must have an element [=stride=] that is a
-multiple of [=RequiredAlignOf=](|T|, |C|) for the storage class |C|:
+Arrays of element type |T| must have an element [=stride=] that is a
+multiple of the [=RequiredAlignOf=](|T|, |C|) for the storage class |C|:
 
 <p algorithm="array element minimum alignment">
     [=StrideOf=](array<|T|[, |N|]>) = |k| &times; [=RequiredAlignOf=](|T|, C)<br>
     Where |k| is a non-negative integer
 </p>
 
+Note: It is not required for [=AlignOf=](|T|) to be a multiple of [=RequiredAlignOf=](|T|, |C|),
+nor for a [=align=] decoration on a member of type |T| to be a multiple of [=RequiredAlignOf=](|T|, |C|),
+so long as each value of type |T| has a byte offset from the start of the structure that is a
+multiple of [=RequiredAlignOf=](|T|, |C|).
 
 The [=storage classes/uniform=] storage class also requires that:
-
 * Array elements are aligned to 16 byte boundaries.
+* The number of bytes between a start of a structure member of a structure type `S` and
```

Comment:
This reads confusingly to me. I think you mean that the member itself has a structure type 'S'.
Suggest:

* If a structure member itself has a structure type `S`, then the number of bytes between the start of that member and the start of any following member must be at least ....

Reply:
Done